core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -357,7 +357,13 @@ private[spark] class BlockManager(
       info: BlockInfo,
       status: BlockStatus,
       droppedMemorySize: Long = 0L): Unit = {
-    val needReregister = !tryToReportBlockStatus(blockId, info, status, droppedMemorySize)
+    def ifAsync: Boolean = {
+      blockId.isBroadcast
+    }
+    // asynchronously report broadcast variables, in future we only need to change the
+    // implementation of ifAsync
```

Comment:
hmmm...I mean, for reporting broadcast, we do it asynchronously, but in future, if we want to report other blocks, we can make ifAsync logic more complicate, e.g.

```

if (blockId.isBroadcast) true
else false

```